beacon-chain/state/getters.go

```diff
@@ -13,36 +15,54 @@ import (
 // EffectiveBalance returns the effective balance of the
 // read only validator.
 func (v *ReadOnlyValidator) EffectiveBalance() uint64 {
+	if v == nil || v.validator == nil {
+		return 0
+	}
 	return v.validator.EffectiveBalance
 }
 
 // ActivationEligibilityEpoch returns the activation eligibility epoch of the
 // read only validator.
 func (v *ReadOnlyValidator) ActivationEligibilityEpoch() uint64 {
+	if v == nil || v.validator == nil {
```

Comment:
i am not sure this is correct, we shouldnt be returning zero values here for `readOnlyValidators`. The correct way is to to check for nil validators when we create the `ReadOnlyValidator` object.

Reply:
Agree, 0 value for uint64s are very dangerous if we aren't careful @shayzluf

Reply:
it is not 0 value it FarFutureEpoch

Reply:
The issue still stands, you are returning a value for a nil validator. `ReadOnlyValidator` is only used in 1 place, so 
we should still be able to check over there, if any nil validators exist and return an error there.

Reply:
its far future epoch and not 0 that is being returned

Reply:
removed it and handled it in ReadOnlyValidator creation